Sources/Leaf/Stem+Spawn.swift

```diff
@@ -19,27 +19,31 @@ extension Stem {
         if name.hasPrefix("/") {
             name = String(name.characters.dropFirst())
         }
-
+        return try spawnLeaf(at: workingDirectory + name)
+    }
+    
+    public func spawnLeaf(at path: String) throws -> Leaf {
+        if let existing = cache?[path] {
+            return existing
+        }
+        
+        let leaf: Leaf
+        
         // non-leaf document. rendered as pure bytes
-        if name.characters.contains("."), !name.hasSuffix(".leaf") {
-            if let existing = cache?[name] { return existing }
-            let path = workingDirectory + name
+        if path.components(separatedBy: "/").last?.contains(".") == true, !path.hasSuffix(".leaf") {
```

Comment:
I'm wondering if this saves much time as opposed to just checking `.suffix(".leaf")` not sure why I had the original `"."` check in there 🤔